modules/core/02-client/keeper/keeper.go

```diff
@@ -355,6 +352,10 @@ func (k *Keeper) IterateClientStates(ctx sdk.Context, storePrefix []byte, cb fun
 	store := ctx.KVStore(k.storeKey)
 	iterator := storetypes.KVStorePrefixIterator(store, host.PrefixedClientStoreKey(storePrefix))
 
+	if cb(exported.LocalhostClientID, localhost.NewClientState(types.GetSelfHeight(ctx))) {
+		return
+	}
```

Comment:
this breaks the contract of the iteration function which should only perform a callback on the prefix provided. If 07-tendermint is the prefix, there should not be a iteration on the 09-localhost